File: react-app/src/assets/enemies.js
export class Enemy {
  constructor(type, damage, health, speed) {
    this.type = type;
    this.damage = damage;
    this.health = health;
    this.speed = speed;
  }
  getStats() {
    return {
      type: this.type,
      damage: this.damage,
      health: this.health,
      speed: this.speed
    };
  }
};

const basicEnemy = new Enemy("basic", 1, 10, 1);
const fastEnemy = new Enemy("fast", 1, 5, 5);
const tankyEnemy = new Enemy("tanky", 5, 50, 0.5);

const enemiesObject = {
  basic: basicEnemy,
  fast: fastEnemy,
  tanky: tankyEnemy
};

export default enemysObject

File: app/seeds/enemies.py
from app.models import db, Enemy
from sqlalchemy.sql import text

def seed_enemies():
    # Assuming these are being placed on path tile_id = 1, 2, 3 for seeding purposes
    enemies = [
        Enemy(
            name="basic",
            damage=1,
            description="A standard enemy with average speed and health.",
            health=10,
            speed=1,
            tile_id=None  # Replace with actual path tile ID
        ),
        Enemy(
            name="fast",
            damage=1,
            description="A fast enemy that moves quickly but is fragile.",
            health=5,
            speed=5,
            tile_id=None  # Replace with actual path tile ID
        ),
        Enemy(
            name="tanky",
            damage=5,
            description="A slow but tough enemy that can take a lot of hits.",
            health=50,
            speed=1,
            tile_id=None  # Replace with actual path tile ID
        )
    ]

    db.session.add_all(enemies)
    db.session.commit()


def undo_enemies():
    if environment == "production":
        db.session.execute(f"TRUNCATE table {SCHEMA}.enemies RESTART IDENTITY CASCADE;")
    else:
        db.session.execute(text("DELETE FROM enemies"))
        
    db.session.commit()